Widget/BusWidget.swift

```diff
@@ -0,0 +1,263 @@
+import FudanKit
+import Intents
+import SwiftUI
+import WidgetKit
+
+@available(iOS 17.0, *)
+struct BusWidgetProvier: AppIntentTimelineProvider {
+    typealias Entry = BusEntry
+    
+    typealias Intent = BusScheduleIntent
+    
+    func placeholder(in context: Context) -> BusEntry {
+        BusEntry()
+    }
+    
+    func snapshot(for configuration: BusScheduleIntent, in context: Context) async -> BusEntry {
+        var entry = BusEntry()
+        let startPoint = configuration.startPoint
+        print(startPoint.rawValue)
+        if !context.isPreview {
+            entry.placeholder = true
+        }
+        return entry
+    }
+    
+    func timeline(for configuration: BusScheduleIntent, in context: Context) async -> Timeline<BusEntry> {
+        do {
+            let (workdayRoutes, holidayRoutes) = try await BusStore.shared.getRefreshedRoutes()
```

Comment:
Loading from internet is unnecessary for data that barely changes. Use local cache for widgets instead.